File: solutions/elixir/darts/1/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @outer_radius 10
  @middle_radius 5
  @inner_radius 1

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position :: position) :: integer
  def score({x, y}) do
    distance = :math.sqrt(:math.pow(x, 2) + :math.pow(y, 2))

    cond do
      distance <= @inner_radius -> 10
      distance <= @middle_radius -> 5
      distance <= @outer_radius -> 1
      true -> 0
    end
  end
end

File: solutions/python/darts/1/darts.py
"""Module for scoring darts"""

import math

OUTER_RIM = 10
MIDDLE_RIM = 5
INNER_RIM = 1

def score(x, y):
    """
    Score a dart throw based on the x, y coodinate relative to the bullseye.

    :param x: int - x coordinate
    :param y: int - y coordinate
    """

    distance = math.sqrt(x ** 2 + y ** 2)

    if OUTER_RIM < distance:
        return 0
    if MIDDLE_RIM < distance <= OUTER_RIM:
        return 1
    if INNER_RIM < distance <= MIDDLE_RIM:
        return 5
    if distance <= INNER_RIM:
        return 10
        
